tests/ClassNameResolver/ClassNameResolverTest.php

```diff
@@ -25,11 +26,15 @@ protected function setUp(): void
     }
 
     #[DataProvider('provideData')]
-    public function test(string $filePath, ClassNames $expectedClassNames): void
+    public function test(string $filePath, ?ClassNames $expectedClassNames): void
     {
         $resolvedClassNames = $this->classNameResolver->resolveFromFromFilePath($filePath);
-        $this->assertInstanceOf(ClassNames::class, $resolvedClassNames);
+        if (!$expectedClassNames instanceof ClassNames) {
+            $this->assertNull($resolvedClassNames);
+            return;
+        }
```

Comment:
I think this should throw an error in this case. We always want a test to provide class.

Reply:
No. Its a valid case as the file beeing analysed does return null.

Reply:
@TomasVotruba does this make sense? would be great to get his merged, I have another edge case I want to fix - waiting for this one :)

Reply:
> I think this should throw an error in this case. We always want a test to provide class.

our test wants to assert that the file beeing analyzed does not return a class (because the only one in the file is annotated `@api`)

Reply:
I see. Conditions in test expectations make it hard to read.

Could you split test method for nulls with own data provider?

Reply:
good idea. just separated the tests